packages/desktop-client/src/components/filters/FilterButtonType.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+
+import { Filter } from '../../icons/v1';
+import { SettingsSliderAlternate } from '../../icons/v2';
+import { theme } from '../../style';
+import Button from '../common/Button';
+import HoverTarget from '../common/HoverTarget';
+import Text from '../common/Text';
+import View from '../common/View';
+import { Tooltip } from '../tooltips';
+
+const ButtonType = cond => {
+  switch (cond) {
+    case 'reports':
+      return <Filter width={15} height={15} />;
+    default:
+      return (
+        <>
+          <SettingsSliderAlternate
+            style={{ width: 16, height: 16, marginRight: 5 }}
+          />{' '}
+          Filter
+        </>
+      );
+  }
+};
+
+type FilterButtonTypeProps = {
+  type: string;
+  onClick: (newValue) => void;
+};
+
+function FilterButtonType({ type, onClick }: FilterButtonTypeProps) {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏this component still has fundamentally the same problem [as mentioned before here](https://github.com/actualbudget/actual/pull/1930#discussion_r1404622816).

What would have been better is to instead have 3x separate components. And have no `type` prop.

Reply:
In order to remove the type prop entirely I'd have to copy/paste everything from the filterbutton function except the icon used. Seems like excessive duplication of code...

Maybe you have something else in mind that I'm not thinking of, but I'm not understanding your vision for this.

Reply:
I took another swing at it in the most recent commit. This one allows for the filter button to be encorporated into the mobile version at some point. LMK!

Reply:
This is way cleaner and more maintainable! Thanks @carkom !